receiver/prometheusreceiver/internal/transaction.go

```diff
@@ -57,17 +68,18 @@ func newTransaction(
 	obsrecv *obsreport.Receiver,
 	trimSuffixes bool) *transaction {
 	return &transaction{
-		ctx:            ctx,
-		families:       make(map[string]*metricFamily),
-		isNew:          true,
-		trimSuffixes:   trimSuffixes,
-		sink:           sink,
-		metricAdjuster: metricAdjuster,
-		externalLabels: externalLabels,
-		logger:         settings.Logger,
-		buildInfo:      settings.BuildInfo,
-		obsrecv:        obsrecv,
-		bufBytes:       make([]byte, 0, 1024),
+		ctx:             ctx,
+		families:        make(map[scopeID]map[string]*metricFamily),
```

Comment:
What is the `string` key used for? If it's just an identifier that doesn't carry any data, I'm curious if a compound key scope+string would be more efficient than the nested map. Did you consider that?

Reply:
The string is the name of the metric family.  So the hierarchy is now `scope -> metric name -> data points`.  Previously it was just `metric name -> data points`

Reply:
I didn't consider that.

Reply:
I understand the hierarchy. I just wanted to see if reducing it makes it any better from performance or readability perspective. I'm not sure about it, so I'm good with any approach